src/include/miopen/oclkernel.hpp

```diff
@@ -155,6 +155,11 @@ class OCLKernel
         }
     }
 
+    OCLKernel(SharedProgramPtr p, const std::string& kernel_name)
+        : program(p), kernel(CreateKernel(p.get(), kernel_name))
+    {
+    }
+
```

Comment:
[Quality] Do not add a new ctor. Just update existing one: read `MIOPEN_DEBUG_COMPILE_ONLY` and skip loading and running the kernels, if enabled.